op-service/sources/l1_beacon_client.go

```diff
@@ -253,6 +253,7 @@ func (cl *L1BeaconClient) GetBlobSidecars(ctx context.Context, ref eth.L1BlockRe
 	}
 
 	if len(hashes) != len(apiscs) {
+		cl.pool.MoveToNext()
```

Comment:
I think the way you had it implemented earlier was better. The fallback logic should stay encapsulated in `fetchSidecars`. This change leaks this out of it (and only makes it then work on the next retry). It's better to have `BeaconBlobSideCars` fail with an error on an empty list, because this is the abstraction of a single `blob_sidecars` request. This makes the fallback logic work again, because it receives a proper error from `BeaconBlobSideCars` on an empty list.

Reply:
The earlier fix `len(resp.Data) < len(hashes)` doesn't **always** work. The beacon api may return more blobs than requested, but not matching the exact blob hashes. I think doing `MoveToNext()` will **always** work with caller side retry.

Reply:
That's true, but I think a case we're not observing right now. We want to fix the case where 0 blobs are returned, while keeping fallback logic working. This solution leaks fallback behavior out of `fetchSidecars` and will then only work on the next retry, which isn't clean.

The ideal solution would be to already test inside `BeaconBlobSideCars` whether all requested hashes are included in the list, but I believe we don't have access to the hashes directly. We do have access to the `Index` though. So best would be to test that the set of all blobhash indices is contained in the returned list, in `BeaconBlobSideCars` already. This is similar to the filtering happening in `GetBlobSidecars`, but only testing for set inclusion.

Reply:
I pushed a new fix which is the combination of the two:)

Reply:
Could you instead implement the index set inclusion, and remove this line here?

Reply:
Ok, pushed, pls take another look!

Reply:
the hash is available for the `IndexedBlobHash` but not in the returned data from the `blob_sidecars` API.